go/bind/extension.go

```diff
@@ -308,30 +319,46 @@ func ExtensionPostJPEG(strConvID, name string, public bool, membersType int,
 		return err
 	}
 
-	// Compute preview result from the native params
-	mimeType := "image/jpeg"
-	location := chat1.NewPreviewLocationWithBytes(previewData)
-	baseMD := chat1.NewAssetMetadataWithImage(chat1.AssetMetadataImage{
-		Width:  baseWidth,
-		Height: baseHeight,
-	})
-	previewMD := chat1.NewAssetMetadataWithImage(chat1.AssetMetadataImage{
-		Width:  previewWidth,
-		Height: previewHeight,
-	})
-	callerPreview := &chat1.MakePreviewRes{
-		MimeType:        mimeType,
-		PreviewMimeType: &mimeType,
-		Location:        &location,
-		Metadata:        &previewMD,
-		BaseMetadata:    &baseMD,
+	var callerPreview *chat1.MakePreviewRes
+	if previewData != nil {
+		// Compute preview result from the native params
+		callerPreview = new(chat1.MakePreviewRes)
+		callerPreview.MimeType = mimeType
+		callerPreview.PreviewMimeType = &mimeType
+		callerPreview.BaseMetadata = new(chat1.AssetMetadata)
+		callerPreview.Metadata = new(chat1.AssetMetadata)
+		location := chat1.NewPreviewLocationWithBytes(previewData)
+		callerPreview.Location = &location
+		switch mimeType {
+		case "image/gif":
+			*callerPreview.BaseMetadata = chat1.NewAssetMetadataWithVideo(chat1.AssetMetadataVideo{
+				Width:      baseWidth,
+				Height:     baseHeight,
+				DurationMs: 10, // make something up, we don't display this anyway
+			})
+			*callerPreview.Metadata = chat1.NewAssetMetadataWithImage(chat1.AssetMetadataImage{
+				Width:  previewWidth,
+				Height: previewHeight,
+			})
+			callerPreview.PreviewMimeType = new(string)
+			*callerPreview.PreviewMimeType = "image/jpeg"
```

Comment:
is image/jpeg correct here?

Reply:
Yeah if we get here, then native has generated a JPEG preview for the GIF. I don't know how to pull a GIF still in native, so I hope this is ok.